src/framework/core/js/FluidRequests.js

```diff
@@ -343,5 +343,205 @@ var fluid_2_0 = fluid_2_0 || {};
+     *       cache. For example pending write requests could be used to service get requests directly.
+     */
+    fluid.defaults("fluid.queuedDataSource", {
+        gradeNames: ["fluid.eventedComponent", "autoInit"],
+        members: {
+            requests: {
+                read: {},
+                write: {}
+            }
+        },
+        components: {
+            wrappedDataSource: {
+                // requires a dataSource that implements the standard set, get, and delete methods.
+                type: "fluid.dataSource"
+            }
+        },
+        events: {
+            enqueued: null,
+            afterRequestComplete: null
+        },
+        listeners: {
+            "enqueued.start": {
+                listener: "{that}.start",
+                args: ["{arguments}.1"]
+            },
+            "afterRequestComplete.start": {
+                listener: "{that}.start",
+                args: ["{arguments}.1"]
+            }
+        },
+        invokers: {
+            set: {
+                funcName: "fluid.queuedDataSource.enqueueWithModel",
+                args: ["{that}", "{that}.requests.write", "{wrappedDataSource}.set", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+            },
+            get: {
+                funcName: "fluid.queuedDataSource.enqueue",
+                args: ["{that}", "{that}.requests.read", "{wrappedDataSource}.get", "{arguments}.0", "{arguments}.1"]
+            },
+            "delete": {
+                funcName: "fluid.queuedDataSource.enqueue",
+                args: ["{that}", "{that}.requests.write", "{wrappedDataSource}.delete", "{arguments}.0", "{arguments}.1"]
+            },
+            start: {
+                funcName: "fluid.queuedDataSource.start",
+                args: ["{that}", "{arguments}.0"]
+            }
+        }
+    });
+
+    fluid.queuedDataSource.start = function (that, queue) {
+        if (!queue.isActive && queue.requests.length) {
+            var request = queue.requests.shift();
+
+            var requestComplete = function () {
+                queue.isActive = false;
+                that.events.afterRequestComplete.fire(request, queue);
+            };
+
+            queue.isActive = true;
+            var args = "model" in request ? [request.directModel, request.model, request.options] : [request.directModel, request.options];
+            var response = request.method.apply(null, args);
+
+            response.then(requestComplete, requestComplete);
+            fluid.promise.follow(response, request.promise);
+        }
+    };
+
+    /*
+     * Adds only one item to the queue at a time, new requests replace older ones
+     *
+     * The request object contains the request function and arguments.
+     * In the form {method: requestFn, directModel: {}, model: {}, callback: callbackFn}
+     */
+    fluid.queuedDataSource.enqueueImpl = function (that, requestsQueue, request) {
+        var promise = fluid.promise();
```

Comment:
See above comment - we need to ensure that the two "enqueue" functions do not have a hard binding to enqueueImpl.